hazelcast/src/main/java/com/hazelcast/config/DiscoveryStrategyConfig.java

```diff
@@ -125,7 +125,6 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(className);
-        out.writeObject(discoveryStrategyFactory);
```

Comment:
can we make `discoveryStrategyFactory` transient then? Or can we completely remove it?

Reply:
I'll add transient. We should leave it as the user might choose to initialise a discovery strategy via a factory and not a class name.